hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/IsStillRunningService.java

```diff
@@ -73,15 +76,56 @@ public boolean isOperationExecuting(Invocation invocation) {
                     invocation.nodeEngine, invocation.serviceName, isStillExecuting,
                     invocation.getTarget(), 0, 0, IS_EXECUTING_CALL_TIMEOUT, null, true);
             Future f = inv.invoke();
-            invocation.logger.warning("Asking if operation execution has been started: " + toString());
+            invocation.logger.warning("Asking if operation execution has been started: " + invocation);
             executing = (Boolean) invocation.nodeEngine.toObject(f.get(IS_EXECUTING_CALL_TIMEOUT, TimeUnit.MILLISECONDS));
         } catch (Exception e) {
-            invocation.logger.warning("While asking 'is-executing': " + toString(), e);
+            invocation.logger.warning("While asking 'is-executing': " + invocation, e);
         }
-        invocation.logger.warning("'is-executing': " + executing + " -> " + toString());
+        invocation.logger.warning("'is-executing': " + executing + " -> " + invocation);
         return executing;
     }
 
+    public void timeoutInvocationIfNotExecuting(final Invocation invocation) {
+        try {
+            final Operation isStillExecuting = createCheckOperation(invocation);
+            final ExecutionCallback callback = new ExecutionCallback() {
```

Comment:
Can you pull out this anonymous inner class. It makes the method very hard to understand.